src/managers/NAVManager.sol

```diff
@@ -0,0 +1,170 @@
+
+    function initializeNetwork(uint16 centrifugeId) external auth {
+        require(accountCounter[centrifugeId] == 0, AlreadyInitialized());
+
+        hub.createAccount(poolId, equityAccount(centrifugeId), false);
+        hub.createAccount(poolId, liabilityAccount(centrifugeId), false);
+        hub.createAccount(poolId, gainAccount(centrifugeId), false);
+        hub.createAccount(poolId, lossAccount(centrifugeId), false);
+
+        accountCounter[centrifugeId] = 5;
+    }
+
+    function initializeHolding(AssetId assetId, IValuation valuation) external auth {
+        uint16 centrifugeId = assetId.centrifugeId();
+        uint16 index = accountCounter[centrifugeId];
+        require(index > 0, NotInitialized());
+        require(index < type(uint16).max, ExceedsMaxAccounts());
+
+        AccountId assetAccount = withCentrifugeId(centrifugeId, index);
+        hub.createAccount(poolId, assetAccount, true);
+        hub.initializeHolding(
+            poolId,
+            scId,
+            assetId,
+            valuation,
+            assetAccount,
+            equityAccount(centrifugeId),
+            gainAccount(centrifugeId),
+            lossAccount(centrifugeId)
+        );
+
+        accountCounter[centrifugeId] = index + 1;
+    }
+
+    function initializeLiability(AssetId assetId, IValuation valuation) external auth {
+        uint16 centrifugeId = assetId.centrifugeId();
+        uint16 index = accountCounter[centrifugeId];
+        require(index > 0, NotInitialized());
+        require(index < type(uint16).max, ExceedsMaxAccounts());
+
+        AccountId expenseAccount = withCentrifugeId(centrifugeId, index);
+        hub.createAccount(poolId, expenseAccount, true);
+        hub.initializeLiability(poolId, scId, assetId, valuation, expenseAccount, liabilityAccount(centrifugeId));
+
+        accountCounter[centrifugeId] = index + 1;
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Price updates
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc ISnapshotHook
+    function onSync(PoolId poolId_, ShareClassId scId_, uint16 centrifugeId) external {
+        require(poolId == poolId_ && scId == scId_);
+        require(msg.sender == holdings, NotAuthorized());
+
+        NetworkMetrics storage networkMetrics = metrics[centrifugeId];
+        D18 netAssetValue_ = netAssetValue(centrifugeId);
+        uint128 issuance = shareClassManager.issuance(scId, centrifugeId);
+
+        globalIssuance = globalIssuance + issuance - networkMetrics.issuance;
+        globalNetAssetValue = globalNetAssetValue + netAssetValue_ - networkMetrics.netAssetValue;
+        D18 price = globalNetAssetValue / d18(globalIssuance);
+
+        networkMetrics.netAssetValue = netAssetValue_;
+        networkMetrics.issuance = issuance;
+
+        hub.updateSharePrice(poolId, scId, price);
```

Comment:
QUESTION

Would we potentially also issue/revoke in this step?
Would we potentially also distribute prices to other domains in this step?